managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2626,6 +2624,9 @@ void internalTrimLedgers(boolean isTruncate, CompletableFuture<?> promise) {
             }
 
             long slowestReaderLedgerId = -1;
+            final long slowestNonDurationLedgerId = getTheSlowestNonDurationReadPosition().getLedgerId();
```

Comment:
We can introduce a new API to the cursor container to get this position and explain  `cursors.getSlowestReaderPosition()` only can get the nonDurable cursor in the future. :)

Reply:
Done